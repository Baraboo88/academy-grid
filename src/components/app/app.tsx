import { ThemeProvider } from 'styled-components';
import {
  Switch,
  Route,
  BrowserRouter as Router,
} from '../common/common';

import DetailedQuest from '../detailed-quest/detailed-quest';
import Contacts from '../contacts/contacts';
import Home from '../home/home';
import { appTheme } from './common';
import * as S from './app.styled';
import React from 'react';
import NotFound from '../not-found/not-found';

const App = () => (
  <ThemeProvider theme={appTheme}>
    <S.GlobalStyle />
    <Router>
      <Switch>
        <Route exact path="/detailed-quest/:id" component={DetailedQuest}/>
        <Route exact path="/contacts">
          <Contacts />
        </Route>
        <Route exact path="/" >
          <Home />
        </Route>
        <Route>
          <NotFound/>
        </Route>
      </Switch>
    </Router>
  </ThemeProvider>
);

export default App;
