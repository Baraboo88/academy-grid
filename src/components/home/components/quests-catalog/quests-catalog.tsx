import { ReactComponent as IconAllQuests } from 'assets/img/icon-all-quests.svg';
import { ReactComponent as IconAdventures } from 'assets/img/icon-adventures.svg';
import { ReactComponent as IconHorrors } from 'assets/img/icon-horrors.svg';
import { ReactComponent as IconMystic } from 'assets/img/icon-mystic.svg';
import { ReactComponent as IconDetective } from 'assets/img/icon-detective.svg';
import { ReactComponent as IconScifi } from 'assets/img/icon-scifi.svg';
import { ReactComponent as IconPerson } from 'assets/img/icon-person.svg';
import { ReactComponent as IconPuzzle } from 'assets/img/icon-puzzle.svg';
import * as S from './quests-catalog.styled';
import {
  applyFilter,
  getCyrillicLevel,
  getCyrillicType,
  QuestModel,
  QuestType,
  StateModel,
} from '../../../../utils/utils';
import { connect } from 'react-redux';
import { Operation } from '../../../../reducer/reducer';
import { getErrorMsg, getIsResponseReceived, getQuests } from '../../../../reducer/selectors';
import React, { useEffect, useState } from 'react';
import { AlertMsg } from '../../home.styled';


interface QuestsCatalogProps {
  quests: QuestModel [];
  onMount: () => void;
  isResponseReceived: boolean;
  errorMsg: string;
}

const QuestsCatalog: React.FC<QuestsCatalogProps> = (props) => {
  const { quests, onMount, isResponseReceived, errorMsg } = props;

  useEffect(() => {
    if (isResponseReceived && errorMsg) {
      setError(errorMsg);
    }
    if (!isResponseReceived) {
      onMount();
    } else {
      setIsLoading(false);
      setInnerQuests(quests);
    }
  }, [quests, getQuests, isResponseReceived, errorMsg]);

  const [error, setError] = useState<string>('');
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [innerQuests, setInnerQuests] = useState<QuestModel []>([]);
  const [activeType, setActiveType] = useState(-1);

  const getTypeIcon = (type: QuestType) => {
    switch (type) {
      case QuestType.Adventures:
        return <IconAdventures />;
      case QuestType.Detective:
        return <IconDetective />;
      case QuestType.Horror:
        return <IconHorrors />;
      case QuestType.Mystic:
        return <IconMystic />;
      case QuestType.SciFi:
        return <IconScifi />;
    }
  };


  return (
    <>
      {error && <AlertMsg>
        {error}
      </AlertMsg>}
      <S.Tabs>
        <S.TabItem>
          <S.TabBtn isActive={activeType === -1} onClick={() => {
            setActiveType(-1);
            setInnerQuests(quests);
          }}>
            <IconAllQuests />
            <S.TabTitle>Все квесты</S.TabTitle>
          </S.TabBtn>
        </S.TabItem>
        {Object.values(QuestType).map((questionType, index) => <S.TabItem key={`${index}-${questionType}`}>
          <S.TabBtn isActive={index === activeType} onClick={() => {
            setActiveType(index);
            setInnerQuests(applyFilter(quests, questionType));
          }}>
            {getTypeIcon(questionType)}
            <S.TabTitle>{getCyrillicType(questionType)}</S.TabTitle>
          </S.TabBtn>
        </S.TabItem>)}

      </S.Tabs>

      <S.QuestsList>

        {!isLoading && innerQuests.map((quest: QuestModel, index) =>
          <S.QuestItem key={`${index}-${quest.id}`}>
            <S.QuestItemLink to={`/detailed-quest/${quest.id}`}>
              <S.Quest>
                <S.QuestImage
                  src={`/${quest.coverImg}`}
                  width='344'
                  height='232'
                  alt='квест Маньяк'
                />

                <S.QuestContent>
                  <S.QuestTitle>{quest.title}</S.QuestTitle>

                  <S.QuestFeatures>
                    <S.QuestFeatureItem>
                      <IconPerson />
                      {`${quest.peopleCount[0]}–${quest.peopleCount[1]} чел`}

                    </S.QuestFeatureItem>
                    <S.QuestFeatureItem>
                      <IconPuzzle />
                      {getCyrillicLevel(quest.level)}
                    </S.QuestFeatureItem>
                  </S.QuestFeatures>
                </S.QuestContent>
              </S.Quest>
            </S.QuestItemLink>
          </S.QuestItem>,
        )}

      </S.QuestsList>
    </>);
};

const mapStateToProps = (state: StateModel) => {

  return {
    quests: getQuests(state),
    errorMsg: getErrorMsg(state),
    isResponseReceived: getIsResponseReceived(state),
  };
};

const mapDispatchToProps = (dispatch: any) => ({
  onMount() {
    dispatch(Operation.getQuests());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(QuestsCatalog);
export {QuestsCatalog};
