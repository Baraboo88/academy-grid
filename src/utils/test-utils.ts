import { QuestLevel, QuestModel, QuestType } from './utils';
import { createApi } from '../api';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';


export const mockQuests: QuestModel[]  = [
    {
      id: 1,
      "title": "Склеп",
      "description": "Средневековое кладбище таит в себе много страшных тайн. Местные жители говорят, что в склепе похоронен граф вампир, который по ночам выходит на охоту, чтобы испить человеческой крови. Через час солнце опустится за горизонт, успеете ли вы убить вампира и выбраться из склепа?",
      "previewImg": "img/preview-sklep.jpg",
      "coverImg": "img/cover-sklep.jpg",
      type: "horror" as QuestType,
      level: "hard" as QuestLevel,
      peopleCount: [2, 5],
      duration: 120
    },
    {
      "id": 2,
      "title": "Маньяк",
      "description": "В комнате с приглушённым светом несколько человек, незнакомых друг с другом, приходят в себя. Никто не помнит, что произошло прошлым вечером. Руки и ноги связаны, но одному из вас получилось освободиться. На стене висит пугающий таймер и запущен отсчёт 60 минут. Сможете ли вы разобраться в стрессовой ситуации, помочь другим, разобраться что произошло и выбраться из комнаты?",
      "previewImg": "img/preview-maniac.jpg",
      "coverImg": "img/cover-maniac.jpg",
      "type": "horror" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [3, 6],
      "duration": 90
    },
    {
      "id": 3,
      "title": "Ритуал",
      "description": "Тяжелый воздух угнетает, в ночи вы оказываетесь запертыми в сыром помещении вместе с другими ничего не понимающими жертвами. Сквозь щель в двери вы видите, как некто в капюшоне готовит площадку как будто для проведения мистического обряда. Удастся ли вам выбраться, пока вы не станете жертвой ритуала?",
      "previewImg": "img/preview-ritual.jpg",
      "coverImg": "img/cover-ritual.jpg",
      "type": "mystic" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [3, 5],
      "duration": 120
    },
    {
      "id": 4,
      "title": "Тайны старого особняка",
      "description": "Погрузитесь в атмосферу служебных помещений закулисья, которые хранят множество тайн и загадок. Вы окажитесь в старом особняке и увидите все, что скрывают его запутанные коридоры.",
      "previewImg": "img/preview-final-frontier.jpg",
      "coverImg": "img/cover-final-frontier.jpg",
      "type": "detective" as QuestType,
      "level": "easy" as QuestLevel,
      "peopleCount": [2, 5],
      "duration": 60
    },
    {
      "id": 5,
      "title": "Хижина в лесу",
      "description": "Вы с друзьями оказались в заброшенной хижине. Какую тайну она скрывает и как из неё выбраться? На эти вопросы вам предстоит найти ответ, чтобы вернуться домой.",
      "previewImg": "img/preview-house-in-the-woods.jpg",
      "coverImg": "img/cover-house-in-the-woods.jpg",
      "type": "mystic" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [4, 7],
      "duration": 90
    },
    {
      "id": 6,
      "title": "Фатальный эксперимент",
      "description": "Вы стоите на пороге нового научного открытия, которое перевернет судьбу человечества. Но что-то идёт не так, и ядерный реактор, который работает на полную мощность, сигнализирует о скорой поломке. Удастся ли вам починить его в отведенное время и предотвратить гибель людей в этом фатальном эксперименте?",
      "previewImg": "img/preview-fatal-exp.jpg",
      "coverImg": "img/cover-fatal-exp.jpg",
      "type": "adventures" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [5, 8],
      "duration": 120
    },

  ]
;


export const getTestStore = () => {
  const initialState = {
    quests: mockQuests,
    currentQuest: mockQuests[1],
    errorMsg: ''
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
