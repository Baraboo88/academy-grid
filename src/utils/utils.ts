import { ActiveTab } from '../reducer/reducer';

export enum QuestType {
  Adventures = 'adventures',
  Horror = 'horror',
  Mystic = 'mystic',
  Detective = 'detective',
  SciFi='sci-fi'
}

export enum QuestLevel {
  Easy = 'easy',
  Medium = 'medium',
  Hard = 'hard'
}

export enum QuestDuration {
  Sixty = 60,
  Ninety = 90,
  OneTwoZero = 120
}

export interface QuestModel {
  id: number;
  title: string;
  description: string;
  previewImg: string;
  coverImg: string;
  type: QuestType;
  level: QuestLevel;
  peopleCount: number [];
  duration: QuestDuration;
}

export interface OrderModel {
  name: string;
  peopleCount: number;
  phone: string;
  isLegal: boolean
}

export interface StateModel {
  quests: QuestModel [];
  currentQuest?: QuestModel;
  isOrderSent?: boolean;
  isResponseReceived: boolean;
  errorMsg: string;
  activeTab: ActiveTab;
}


export const getCyrillicLevel = (level: QuestLevel) => {
  switch (level) {
    case QuestLevel.Easy:
      return 'простой';
    case QuestLevel.Medium:
      return 'средний';
    case QuestLevel.Hard:
      return 'сложный';
  }
  return '';
};

export const getCyrillicType = (questType: QuestType) => {
  switch (questType) {
    case QuestType.Adventures:
      return 'Приключения';
    case QuestType.Horror:
      return 'Ужасы';
    case QuestType.Mystic:
      return 'Мистика';
    case QuestType.Detective:
      return 'Детектив';
    case QuestType.SciFi:
      return 'Sci-fi';
  }
  return '';
};


export const applyFilter = (quests: QuestModel[], filter: QuestType) => quests.filter((quest) => quest.type === filter);
