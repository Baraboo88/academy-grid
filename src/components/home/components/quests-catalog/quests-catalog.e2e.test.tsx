import * as Enzyme from 'enzyme';
import { mount } from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';

import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import { getTestStore, mockQuests } from '../../../../utils/test-utils';
import { appTheme } from '../../../app/common';

import { BrowserRouter } from 'react-router-dom';
import { QuestsCatalog } from './quests-catalog';


Enzyme.configure({ adapter: new EnzymeReactAdapter() });


describe(`QuestsCatalog e2e`, () => {
  const mockOnMount = jest.fn();

  beforeEach(() => {

    mount(<Provider store={getTestStore()}><BrowserRouter><ThemeProvider
      theme={appTheme}><QuestsCatalog quests={mockQuests}
                                      onMount={mockOnMount}
                                      isResponseReceived={false}
                                      errorMsg={''} /></ThemeProvider></BrowserRouter></Provider>);
  })


  it(`Should onMount successfully working`, () => {
    expect(mockOnMount).toHaveBeenCalledTimes(1);
    expect(mockOnMount).toHaveBeenCalledWith();
  });

});
