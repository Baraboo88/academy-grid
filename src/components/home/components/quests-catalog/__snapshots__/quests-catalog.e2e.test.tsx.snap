// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingModal successfully rendered 1`] = `
Array [
  "",
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0;
  margin-bottom: 20px;
  padding: 0;
  list-style: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 40px;
  max-width: 190px;
  margin-bottom: 20px;
}

.c1:not(:last-of-type) {
  padding-right: 40px;
  margin-right: 39px;
  border-right: 1px solid rgba(255,255,255,0.32);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  font-family: inherit;
  font-size: 14px;
  line-height: 140%;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  font-weight: 700;
  color: #E6E6E6;
  background: transparent;
  border: none;
  cursor: pointer;
}

.c2 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c2:focus span,
.c2:hover span {
  border-bottom: 2px solid #F2890F;
}

.c2 span {
  border-bottom: 2px solid #F2890F;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  font-family: inherit;
  font-size: 14px;
  line-height: 140%;
  -webkit-letter-spacing: -0.02em;
  -moz-letter-spacing: -0.02em;
  -ms-letter-spacing: -0.02em;
  letter-spacing: -0.02em;
  font-weight: 700;
  color: #E6E6E6;
  background: transparent;
  border: none;
  cursor: pointer;
}

.c4 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c4:focus span,
.c4:hover span {
  border-bottom: 2px solid #F2890F;
}

.c3 {
  margin-left: 13px;
  padding-top: 4px;
  padding-bottom: 3px;
  border-bottom: 2px solid transparent;
}

@media (max-width:1110px) {
  .c0 {
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
  }
}

@media (max-width:1110px) {
  .c1:not(:last-of-type) {
    padding-right: 30px;
    margin-right: 29px;
  }
}

@media (max-width:1150px) {

}

@media (max-width:1150px) {

}

<ul
    className="c0"
  >
    <li
      className="c1"
    >
      <button
        className="c2"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-all-quests.svg
        </svg>
        <span
          className="c3"
        >
          Все квесты
        </span>
      </button>
    </li>
    <li
      className="c1"
    >
      <button
        className="c4"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-adventures.svg
        </svg>
        <span
          className="c3"
        >
          Приключения
        </span>
      </button>
    </li>
    <li
      className="c1"
    >
      <button
        className="c4"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-horrors.svg
        </svg>
        <span
          className="c3"
        >
          Ужасы
        </span>
      </button>
    </li>
    <li
      className="c1"
    >
      <button
        className="c4"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-mystic.svg
        </svg>
        <span
          className="c3"
        >
          Мистика
        </span>
      </button>
    </li>
    <li
      className="c1"
    >
      <button
        className="c4"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-detective.svg
        </svg>
        <span
          className="c3"
        >
          Детектив
        </span>
      </button>
    </li>
    <li
      className="c1"
    >
      <button
        className="c4"
        onClick={[Function]}
        type="button"
      >
        <svg>
          icon-scifi.svg
        </svg>
        <span
          className="c3"
        >
          Sci-fi
        </span>
      </button>
    </li>
  </ul>,
  .c0 {
  display: grid;
  grid-template-columns: repeat(3,344px);
  gap: 32px 24px;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-width: 344px;
  min-height: 232px;
}

.c2::after {
  content: '';
  z-index: 2;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 160px;
  background-image: linear-gradient( 0.36deg, rgba(28,27,27,0.9) 5.23%, rgba(46,46,46,0) 98.38% );
  border-radius: 3px;
}

.c4 {
  position: relative;
  z-index: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  margin-top: auto;
}

.c5 {
  max-width: 264px;
  max-height: 59px;
  margin: 0;
  margin-bottom: 16px;
  padding: 0;
  font-size: 24px;
  line-height: 120%;
  font-weight: 700;
  color: #FFFFFF;
  word-break: break-word;
  overflow: hidden;
}

.c3 {
  z-index: 1;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 4px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0;
  list-style: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 20px;
  max-width: 100px;
  max-height: 50px;
  word-break: break-word;
  overflow: hidden;
  font-size: 13px;
  line-height: 144%;
  color: #E6E6E6;
}

.c7:not(:last-of-type) {
  padding-right: 13px;
  margin-right: 12px;
  border-right: 1px solid rgba(255,255,255,0.5);
}

.c7 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-right: 9px;
  margin-bottom: 2px;
}

@media (max-width:1110px) {

}

@media (max-width:1110px) {

}

@media (max-width:1150px) {
  .c0 {
    grid-template-columns: repeat(3,minmax(303px,1fr));
  }
}

@media (max-width:1150px) {
  .c2 {
    min-width: 303px;
    min-height: 204px;
  }
}

<ul
    className="c0"
  >
    <li
      className=""
    >
      <a
        className="c1"
        href="/detailed-quest/1"
        onClick={[Function]}
      >
        <article
          className="c2"
        >
          <img
            alt="квест Маньяк"
            className="c3"
            height="232"
            src="/img/cover-sklep.jpg"
            width="344"
          />
          <div
            className="c4"
          >
            <h2
              className="c5"
            >
              Склеп
            </h2>
            <ul
              className="c6"
            >
              <li
                className="c7"
              >
                <svg>
                  icon-person.svg
                </svg>
                2–5 чел
              </li>
              <li
                className="c7"
              >
                <svg>
                  icon-puzzle.svg
                </svg>
                сложный
              </li>
            </ul>
          </div>
        </article>
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="c1"
        href="/detailed-quest/2"
        onClick={[Function]}
      >
        <article
          className="c2"
        >
          <img
            alt="квест Маньяк"
            className="c3"
            height="232"
            src="/img/cover-maniac.jpg"
            width="344"
          />
          <div
            className="c4"
          >
            <h2
              className="c5"
            >
              Маньяк
            </h2>
            <ul
              className="c6"
            >
              <li
                className="c7"
              >
                <svg>
                  icon-person.svg
                </svg>
                3–6 чел
              </li>
              <li
                className="c7"
              >
                <svg>
                  icon-puzzle.svg
                </svg>
                средний
              </li>
            </ul>
          </div>
        </article>
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="c1"
        href="/detailed-quest/3"
        onClick={[Function]}
      >
        <article
          className="c2"
        >
          <img
            alt="квест Маньяк"
            className="c3"
            height="232"
            src="/img/cover-ritual.jpg"
            width="344"
          />
          <div
            className="c4"
          >
            <h2
              className="c5"
            >
              Ритуал
            </h2>
            <ul
              className="c6"
            >
              <li
                className="c7"
              >
                <svg>
                  icon-person.svg
                </svg>
                3–5 чел
              </li>
              <li
                className="c7"
              >
                <svg>
                  icon-puzzle.svg
                </svg>
                сложный
              </li>
            </ul>
          </div>
        </article>
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="c1"
        href="/detailed-quest/4"
        onClick={[Function]}
      >
        <article
          className="c2"
        >
          <img
            alt="квест Маньяк"
            className="c3"
            height="232"
            src="/img/cover-final-frontier.jpg"
            width="344"
          />
          <div
            className="c4"
          >
            <h2
              className="c5"
            >
              Тайны старого особняка
            </h2>
            <ul
              className="c6"
            >
              <li
                className="c7"
              >
                <svg>
                  icon-person.svg
                </svg>
                2–5 чел
              </li>
              <li
                className="c7"
              >
                <svg>
                  icon-puzzle.svg
                </svg>
                простой
              </li>
            </ul>
          </div>
        </article>
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="c1"
        href="/detailed-quest/5"
        onClick={[Function]}
      >
        <article
          className="c2"
        >
          <img
            alt="квест Маньяк"
            className="c3"
            height="232"
            src="/img/cover-house-in-the-woods.jpg"
            width="344"
          />
          <div
            className="c4"
          >
            <h2
              className="c5"
            >
              Хижина в лесу
            </h2>
            <ul
              className="c6"
            >
              <li
                className="c7"
              >
                <svg>
                  icon-person.svg
                </svg>
                4–7 чел
              </li>
              <li
                className="c7"
              >
                <svg>
                  icon-puzzle.svg
                </svg>
                средний
              </li>
            </ul>
          </div>
        </article>
      </a>
    </li>
    <li
      className=""
    >
      <a
        className="c1"
        href="/detailed-quest/6"
        onClick={[Function]}
      >
        <article
          className="c2"
        >
          <img
            alt="квест Маньяк"
            className="c3"
            height="232"
            src="/img/cover-fatal-exp.jpg"
            width="344"
          />
          <div
            className="c4"
          >
            <h2
              className="c5"
            >
              Фатальный эксперимент
            </h2>
            <ul
              className="c6"
            >
              <li
                className="c7"
              >
                <svg>
                  icon-person.svg
                </svg>
                5–8 чел
              </li>
              <li
                className="c7"
              >
                <svg>
                  icon-puzzle.svg
                </svg>
                сложный
              </li>
            </ul>
          </div>
        </article>
      </a>
    </li>
  </ul>,
]
`;
